crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -135,12 +140,133 @@ impl Persistence {
             .await
             .map_err(Error::DbError)
     }
+
+    /// Get auction data.
+    pub async fn get_auction(
+        &self,
+        auction_id: domain::auction::Id,
+    ) -> Result<domain::settlement::Auction, error::Auction> {
```

Comment:
DB metrics should probably be added for each SQL query inside this function.

Reply:
timer?

Reply:
Yep

Reply:
added one for the whole function, will add more later if it takes too long